opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -1078,6 +1101,14 @@ namespace detail {
                 const Eigen::VectorXd& dx = solver.solve(total_residual_v.matrix());
                 assert(dx.size() == total_residual_v.size());
                 asImpl().wellModel().updateWellState(dx.array(), dpMaxRel(), well_state);
+
+                // Enforce the VREP control
```

Comment:
This section is repeated here, why? If it really needs to be repeated it should be factored into a small method of its own.